### PR TITLE
Feature/77 aggregate threads from connections

### PR DESCRIPTION
**Pull request by davideastmond**
- Created method that grabs all the threads that were created by the source user's connections and puts them in an array, sorted by most recent.

- Implemented the GET `/threads` route. The request to the route returns an object that has the array mentioned above. We can build on this object to include other data as we expand this functionality out.

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "cookie-session": "^1.4.0",
     "cors": "^2.8.5",
     "crypto-js": "^4.0.0",
+    "dayjs": "^1.9.6",
```
Comment (davideastmond) [inline api/package.json:26]: Not sure we'll need this as of yet, but I installed it anyway

File: api/src/models/user/user.methods.ts [hunk 2]
```diff
@@ -198,22 +198,48 @@ export async function createAndPostThread(this: IUserDocument, threadDetails: IT
...
+  const threads: IThread[] = [];
+
+  users.forEach((user) => {
+    for (const[_, value] of Object.entries(user.threads.started)) {
+      threads.push(value);
+    }
+  });
 
+  // Sort the threads and return
+  return threads.sort((a, b) => {
```
Comment (lily-law) [inline api/src/models/user/user.methods.ts:233]: This could be simplified 
return threads.sort((a, b) => b.createdAt.valueOf() - a.createdAt.valueOf());

**Review by lily-law — APPROVED** [review]
This is looking good 🚀 

User methods is growing a little big. There looks to be 4 distinct parts to it: auth, profile, connections, and threads.

File: api/src/models/user/user.methods.ts (hunk 2 above)
Comment (davideastmond) [reply api/src/models/user/user.methods.ts:233]: I was testing out sorting dates in the node SLI and it wasn't sorting properly with this syntax. But I updated it and tests are passing, so good!

**Pull request merged**